### ISSUE TITLE
Restrictions on Interleaved Columns

### ISSUE THREAD

**Issue by DavePearce** [Consensys/go-corset#206]
The `definterleaved` constraint form has a number of quite specific restrictions on its use:

* Number of columns must be `2^n` (i.e. `2`, `4`, `8`, etc but not `3` or `5`)
* Cannot interleave already interleaved columns.
* Cannot be used in vanishing constraints.
* Can only be used as source for sorted permutations.

### Approach

1. Add some `invalid` test cases capturing the above.  Potentially, some existing tests are now invalid.
1. (`pkg/corset/compiler/Parser.java`) If the number of columns is not a power of `2`, return a `SyntaxError` stating this.
2. (`pkg/corset/compiler/Resolver.java`) Extend `LocalScope` to report `SyntaxError` for any column access where `multiplier > 1`.
3. (`pkg/corset/compiler/Resolver.java`) in `finaliseDefInterleaving()` should also check that source columns have `multiplier == 1`.'

@AlexandreBelling Any comments on this?  Just want to make sure these rules are still valid.

**Comment by AlexandreBelling** [comment]
Yes, these are still valid rules. Ok for saying definterleaved may only be used for permutation constraints.
